engine/src/cubos/engine/gizmos/plugin.cpp

```diff
@@ -190,13 +190,26 @@ static void drawGizmosSystem(Write<Gizmos> gizmos, Write<GizmosRenderer> gizmosR
                              Query<Read<LocalToWorld>, Read<Camera>> query)
 {
     auto screenSize = (*window)->framebufferSize();
+    gizmosRenderer->renderDevice->setShaderPipeline(gizmosRenderer->drawPipeline);
+    gizmosRenderer->renderDevice->clearColor(0, 0, 0, 0);
+
+    cubos::core::gl::DepthStencilStateDesc dss;
+    dss.depth.enabled = true;
+    dss.depth.writeEnabled = true;
+    gizmosRenderer->renderDevice->setDepthStencilState(gizmosRenderer->renderDevice->createDepthStencilState(dss));
```

Comment:
Create the depth stencil states at initialization instead of creating a new one every frame.